tests/attribute_tests.cpp

```diff
@@ -401,4 +401,48 @@ BOOST_AUTO_TEST_CASE(BytesClone)
     BOOST_CHECK_EQUAL(cs->type(), 19);
 }
 
+BOOST_AUTO_TEST_CASE(ChapPasswordDataConstructor)
+{
+    std::vector<uint8_t> d {'1', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f', 'g'};
```

Comment:
Why do you use chars here and hex codes below? It would be clearer to use either chars or hexes in both cases.